transformer/test_transform.py

```diff
@@ -13,8 +14,18 @@ def test_it_returns_a_locustfile_string_given_scenario_path(self, tmp_path: Path
         har_path = tmp_path / "some.har"
         with har_path.open("w") as file:
             json.dump(_DUMMY_HAR_DICT, file)
-        locust_file = transformer.transform.main(har_path)
+        locust_file = main(har_path)
         try:
             compile(locust_file, "locustfile.py", "exec")
         except Exception as exception:
             pytest.fail(f"Compiling locustfile failed. [{exception}].")
+
+    @patch("transformer.transform.locustfile")
+    @patch("transformer.transform.Scenario.from_path")
+    def test_it_uses_default_plugins(self, scenario_from_path, _, tmp_path: Path):
+        har_path = tmp_path / "some.har"
+        with har_path.open("w") as file:
+            json.dump(_DUMMY_HAR_DICT, file)
+
+        main(har_path)
+        scenario_from_path.assert_called_once_with(har_path, DEFAULT_PLUGINS)
```

Comment:
Could we test the same thing without relying on our knowledge of `locustfile` and `Scenario.from_path`? I think we could for example do something like:
```py
# removed: @patch
def test_it_uses_default_plugins(self, tmp_path: Path, monkeypatch):  # added: monkeypatch
    times_plugin_called = 0

    def fake_plugin(tasks):
        nonlocal times_plugin_called
        times_plugin_called += 1
        return tasks

    monkeypatch.setattr(transformer.transform, "DEFAULT_PLUGINS", [fake_plugin])

    main(har_path, plugins=[])  # explicitly provide no plugins

    assert times_plugin_called == 1
```

This would avoid mocking the call hierarchy below `main` (allowing it to change) and also to run actual plugins (that could have side effects). What do you think about this?

Reply:
Discussed with Brian and implemented.